src/pull_requests.py

```diff
@@ -21,29 +29,42 @@ def fetch_pull_requests(pull_request_query: PullRequestQuery) -> list[PullReques
     if pull_request_query.title:
         filter_params += f" in:title {pull_request_query.title}"
 
-    issues = g.search_issues(query=filter_params)
-
-    pull_requests = []
-    for issue in issues:
-        if issue.pull_request:
-            pr = issue.as_pull_request()
-            pull_requests.append(pr)
-
-    return pull_requests
+    issues = gh.search_issues(query=filter_params)
+    return list(_fetch_prs_concurrently(issues))
 
 
 def fetch_updated_pull_requests(
     current_pull_requests: list[PullRequest], modified_pull_requests: list[PullRequest]
 ) -> list[PullRequest]:
-    g = Github(st.session_state.token)
-    updated_pull_requests = current_pull_requests.copy()
-
-    for i, pr in enumerate(updated_pull_requests):
-        for modified_pr in modified_pull_requests:
-            if pr.number == modified_pr.number:
-                repo = g.get_repo(full_name_or_id=pr.base.repo.full_name)
-                updated_pr = repo.get_pull(number=pr.number)
-                updated_pull_requests[i] = updated_pr
-                break
-
-    return current_pull_requests
+    gh = Github(st.session_state.token)
+    updated_prs = _fetch_updated_prs_concurrently(gh, modified_pull_requests)
+    return list(_merge_pr_lists(current_pull_requests, updated_prs))
+
+
+def _fetch_prs_concurrently(
+    issues: Iterable[Issue],
+    *,
+    executor: Executor = DEFAULT_EXECUTOR,
+) -> Iterable[PullRequest]:
+    with executor as task_pool:
+        return task_pool.map(Issue.as_pull_request, issues)
+
+
+def _fetch_updated_prs_concurrently(
+    gh: Github,
+    prs: Iterable[PullRequest],
+    *,
+    executor: Executor = DEFAULT_EXECUTOR,
+) -> Iterable[PullRequest]:
+    with executor as task_pool:
+        return task_pool.map(partial(_fetch_updated_pr, gh), prs)
+
+
+def _fetch_updated_pr(gh: Github, pr: PullRequest) -> PullRequest:
+    return gh.get_repo(pr.base.repo.full_name).get_pull(pr.number)
+
+
+def _merge_pr_lists(a: Iterable[PullRequest], b: Iterable[PullRequest]) -> Iterable[PullRequest]:
```

Comment:
Nitpick: The name here can be confusing as it seems we are merging the prs. 😄

Reply:
Hahaha, good catch, I'll try to come up with a better one